File: Algo/hr_implement/2/cumulative_viral_ads.js
function viralAdvertising(n){
    let ads=5,
    likes=0,
    refer = 3;

    for(let i=1;i<=n;i++){
        let tmp_like = Math.floor(ads/2);
        likes += tmp_like;
        ads = tmp_like*refer;
    }
    return likes;
}

console.log('likes',viralAdvertising(3));

File: Algo/hr_implement/2/cumulative_viral_ads.cs
using System;
using System.Linq;
using System.Collections;

public class Test
{
    public void TestFun()
    {
       int n =3,likes=0,refer=3;;
       double ads=5;

       for(int i=1;i<=n;i++)
       {
           int tmp_like = (int) Math.Floor(ads/2);
           likes += tmp_like;
           ads = tmp_like * refer;
       }

       Console.WriteLine(likes);
    }
}